### ISSUE TITLE
[rush] pnpm Blocked by Group Policy

### ISSUE THREAD

**Issue by sheldonnofer** [microsoft/rushstack#2187]
**Is this a feature or a bug?**

- [ ] Feature
- [x] Bug

**Please describe the actual behavior.**
When cloning a Rush repo and running `rush update` on a corporate Windows 10 computer, the `pnpm install` command fails due to our org's group policy. We solve this issue with npm by setting the prefix parameter in `.npmrc`(https://docs.npmjs.com/using-npm/config#prefix), and then all global npm executables are placed in a location that works with our group policy. Here is a sample of the output from `rush update`:

```bash
Running "pnpm install" in C:\SRC\workspace\project\common\temp

This program is blocked by group policy. For more information, contact your system administrator.

The command failed:
 C:\SRC\workspace\project\common\temp\pnpm-local\node_modules\.bin\pnpm install --store C:\SRC\workspace\project\common\temp\pnpm-store --no-lock --no-prefer-frozen-lockfile --resolution-strategy=fewer-dependencies
ERROR: Error: The command failed with exit code 1
```

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**
I am not an IT expert, just a programmer. Our org only allows scripts, executables, etc. to be ran from certain locations, such as `C:\SRC`. So this is a specific problem with our corp. However, these would be the steps to reproduce:
* Use a corporate Windows 10 computer with similar group policies
* Clone a Rush repo
* Run `rush update`

**What is the expected behavior?**
`rush update` should install all the dependencies for our apps/packages.

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** Rush
* **Tool Version:**  5.33.2
* **Node Version:** 12.18.3
  * **Is this a LTS version?** Yes
  * **Have you tested on a LTS version?** Yes
* **OS:** Windows 10 Version 1909 (OS Build 18363.1016)

**Potential solution**
It seems like allowing the configuration of where rush looks for a pnpm binary would be all that is required to solve this issue. For example: 
https://github.com/microsoft/rushstack/blob/2b06f26d1f7a67351a58adea2c1bff1f0fbe09cf/apps/rush-lib/src/api/RushGlobalFolder.ts#L33
If this line looked for an environment variable or some other parameter set in rush.json instead of always resolving to the user's homedir, we could tell it to save the `.rush` directory in a location not blocked by group policy.

CC: @octogonz

**Comment by octogonz** [comment]
An environment variable does seem like a good solution for this problem.  Normally we are concerned about determinism (e.g. what if different VMs or shell prompts have different environment variables?), but in this case the Rush user folder is a cache.  So the worst that would happen would be multiple copies of the cache.

Other similar paths are [already configurable](https://rushjs.io/pages/configs/environment_vars/) via `RUSH_TEMP_FOLDER` and `RUSH_PNPM_STORE_PATH`.

**Comment by octogonz** [comment]
Since the settings are managed by `RushGlobalFolder.ts`, I'm thinking the environment variable would be `RUSH_GLOBAL_FOLDER`.

This might be an easy fix -- lemme see if I can make a PR.

**Comment by octogonz** [comment]
Here's a PR: https://github.com/microsoft/rushstack/pull/2188

**Comment by octogonz** [comment]
This was released with Rush 5.34.1.
